framework/auth/views.py

```diff
@@ -637,3 +639,79 @@ def resend_confirmation_post(auth):
 
     # Don't go anywhere
     return {'form': form}
+
+
+def oauth_user_email_get():
+    """
+    Landing view for first-time oauth-login user to enter their email address.
+    HTTP Method: GET
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_oauth_first_time_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    oauth_provider = session.data['oauth_user_provider']
+
+    return {
+        'form': form,
+        'oauth_provider': oauth_provider.upper()
+    }
+
+
+def oauth_user_email_post():
+    """
+    View to handle email submission for first-time oauth-login user.
+    HTTP Method: POST
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_oauth_first_time_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    oauth_provider = session.data['oauth_user_provider']
+    oauth_id = session.data['oauth_user_id']
+    oauth_fullname = session.data['oauth_user_fullname']
+
+    if form.validate():
+        clean_email = form.email.data
+        user = get_user(email=clean_email)
+        oauth = {oauth_provider: oauth_id}
+        if user:
+            # TODO: link user's OSF account with ORCID
+            # 1. update user oauth
+            # 2. send confirmation email
+            # 3. notify user
+            # 4. remove session and osf cookie
+            pass
+        else:
+            # TODO: create a new account for the user
+            # 1. create unconfirmed user with oauth
+            # 2. update social fields if oauth provider in social
+            # 3. send confirmation email
+            # 4. notify user
+            # 5. remove session and osf cookie
+            user = User.create_unconfirmed(
+                username=clean_email,
+                password=str(uuid.uuid4()),
+                fullname=oauth_fullname,
+                oauth=oauth,
+                campaign=None
+            )
+            # TODO: update social fields
+            user.save()
+            framework_auth.signals.user_registered.send(user)
+            send_confirm_email(user, email=user.username)
```

Comment:
Consider modifying `send_confirm_email` to determine that these events are OAuth related, and to change the email template accordingly